workspaces/src/operations.rs

```diff
@@ -276,11 +279,12 @@ impl<'a, 'b, T: Network> CallTransaction<'a, 'b, T> {
                 self.function.deposit,
             )
             .await
+            .map_err(crate::error::Error::from)
             .and_then(CallExecutionDetails::from_outcome)
```

Comment:
If I am reading this correctly, this'll convert faield transactions into errors, erasing all information except for status. I think this is wrong: "the RPC failed due to IO error" and "transaction execution itself failed due to contract returning with error" seems to be very dissimilar error conditions. 

This should be `map(CallExecutionDetails::from)` rather than `and_then(from_outcome)`

Reply:
this was to force users to handle the failed transaction instead of ignoring them, but definitely loses info. I then learned about `#[muse_use]` a while back and went into this new direction with this PR: https://github.com/near/workspaces-rs/pull/150 to handle this case specifically